src/controllers/authController.js

```diff
@@ -1,94 +1,57 @@
 const User = require('../models/User');
-const jwt = require('jsonwebtoken');
 const argon2 = require('argon2');
-
-// TODO: SET up CUSTOM ERRORs later
-
-// create token
-const createJWT = (user) => {
-  const payload = {
-    name: user.name,
-    userId: user._id,
-    role: user.role,
-  };
-  const token = jwt.sign(payload, process.env.JWT_SECRET, {
-    expiresIn: process.env.JWT_LIFETIME,
-  });
-  return token;
-};
+const { StatusCodes } = require('http-status-codes');
+const CustomError = require('../errors');
+const { attachCookiesToResponse, createTokenUser } = require('../utils');
 
 const register = async (req, res) => {
-  const { name, email, password } = req.body;
-
-  const emailAlreadyExists = await User.findOne({ email });
+  const { email, name, password } = req.body; // Extract data from the request
 
+  const emailAlreadyExists = await User.findOne({ email }); // Check if a user with the email already exists
   if (emailAlreadyExists) {
-    res.status(400).json({
-      msg: 'This email already exists in a database',
-    });
+    throw new CustomError.BadRequestError('Email already exists'); // If user exists, throw an error
   }
 
-  //first registered user is an admin
-  //get all users, and if there are no users assign role to admin
-  const isFirstAccount = (await User.countDocuments({})) === 0;
-  const role = isFirstAccount ? 'admin' : 'user';
+  const isFirstAccount = (await User.countDocuments({})) === 0; // Check if it's the first account
+  const role = isFirstAccount ? 'admin' : 'user'; // Assign a role based on first account or not
 
-  const hashedPassword = await argon2.hash(password);
+  const hashedPassword = await argon2.hash(password); // Hash the password
```

Comment:
In the User model, we currently have a half-finished pre-save hook, where this password hashing could occur: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/2de5af21b18d62bce63c37223da0e5a92378c8de/src/models/User.js#L62-L69](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/2de5af21b18d62bce63c37223da0e5a92378c8de/src/models/User.js#L62-L69)

If you prefer to instead do it in-line here in the controller, that's totally fine, but in that case would you mind using this PR to delete that pre-save hook in the model since it's not doing anything now apart from logging to console?

Reply:
I've moved password hashing to User.js and removed it from controllers as discussed. I'll handle any conflicts in PS27 in favor of User.js.
<img width="559" alt="Снимок экрана 2023-08-09 190413" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/bdff83ec-3da7-441f-b2d5-70a605604d68">